core/src/data/Geometry.cpp

```diff
@@ -98,39 +98,47 @@ void Geometry::generatePositions()
     std::int64_t max_b = std::min( 10, n_cells[1] );
     std::int64_t max_c = std::min( 10, n_cells[2] );
     Vector3 diff;
-    for( std::int64_t i = 0; i < n_cell_atoms; ++i )
+
+    // This scales quadratically in n_cell_atoms, so we check for co-incident positions only if n_cell_atoms is somewhat small!
+    if( n_cell_atoms < 100 )
```

Comment:
this needs an `else` with a warning log message telling the user that only the first N atoms were checked